pkg/common/helpers.go

```diff
@@ -48,6 +49,14 @@ func GetTimestampString() string {
 	return strconv.FormatInt(GetTimestamp(), 10)
 }
 
+// InitUUID enables randomness pool setting for UUID package
+// This method is not threadsafe and must be called in the same package context that calls other
+// uuid methods
+// This will get called once on initialisation to ensure randomness of ID generation
+func InitUUID() {
+	uuid.EnableRandPool()
```

Comment:
This might be overthinking it, but is there any risk of this being called multiple times in case users initialize more than one tracker? Would it make sense to have a global flag to indicate if it was already called or not?

Reply:
Good point! 

This made me think about it and experiment a bit - I committed a workaround where it only gets called on initialisation.

I haven't got much time to test very thoroughly, but I tested the method as follows:

```
package mypkg

import "fmt"

func PrintSomething() string {
	fmt.Println("PrintSomething WAS CALLED")

	return "This string is just to avoid go compiler error"
}

var _ = PrintSomething()

func DoSomething() {
	fmt.Println("DoSomething WAS CALLED")
}
```

```
package otherpkg

import (
	"fmt"
	mypkg "getOneShard/testpkg"
)

func DoSomethingElse() {
	fmt.Println("DoSomethingElse was called")
	mypkg.DoSomething()
}
```

```

func main() {
	mypkg.DoSomething()
	mypkg.DoSomething()
	mypkg.DoSomething()

	otherpkg.DoSomethingElse()
```

Result: 
```
% go run main.go
PrintSomething WAS CALLED
DoSomething WAS CALLED
DoSomething WAS CALLED
DoSomething WAS CALLED
DoSomethingElse was called
DoSomething WAS CALLED
```

I don't know if this is an acceptable solution though, or even if I like it very much

Reply:
I like this approach, makes sense!

I'll try to do a bit more testing and merge it in.

Reply:
I didn't think of that!

I think that's a nice compromise, but also I wouldn't even suggest that a compromise is necessary. In reality UUID conflicts are very rare and the difference here is a ~0.5% improvement.

This may well be a solution without a problem - up to you ofc, but I'm not in favour of adding options that don't have requirements, so I wouldn't be put out by deciding just to shelf this unless we actually hit a reason to do so.

Reply:
PS. Thanks for the care and attention!